src/SFML/Window/GlContext.cpp

```diff
@@ -582,18 +603,30 @@ void GlContext::initialize(const ContextSettings& requestedSettings)
     else
     {
         // Try the old way
-        const GLubyte* version = glGetString(GL_VERSION);
+
+        // If we can't get the version number, assume 1.1
+        m_settings.majorVersion = 1;
+        m_settings.minorVersion = 1;
+
+        const char* version = reinterpret_cast<const char*>(glGetString(GL_VERSION));
         if (version)
         {
-            // The beginning of the returned string is "major.minor" (this is standard)
-            m_settings.majorVersion = version[0] - '0';
-            m_settings.minorVersion = version[2] - '0';
+            // OpenGL ES Common Lite profile: The beginning of the returned string is "OpenGL ES-CL major.minor"
+            // OpenGL ES Common profile:      The beginning of the returned string is "OpenGL ES-CM major.minor"
+            // OpenGL ES Full profile:        The beginning of the returned string is "OpenGL ES major.minor"
+            // Desktop OpenGL:                The beginning of the returned string is "major.minor"
+
+            if (!parseVersionString(version, "OpenGL ES-CL ", m_settings.majorVersion, m_settings.minorVersion) &&
+                !parseVersionString(version, "OpenGL ES-CM ", m_settings.majorVersion, m_settings.minorVersion) &&
+                !parseVersionString(version, "OpenGL ES ",    m_settings.majorVersion, m_settings.minorVersion) &&
+                !parseVersionString(version, "",              m_settings.majorVersion, m_settings.minorVersion))
```

Comment:
Sorry for the minor remark, but logical ors || make more sense here, don't they?

Reply:
!A && !B <=> A || B

Since you want to check that these are not the values you expect, not-anding seems more logical to me.